apps/graphql-playground/src/components/GqlPlayground.tsx

```diff
@@ -83,17 +88,41 @@ function GqlPlayground(props: GqlPlaygroundProps) {
         },
       ];
 
-  const settings = { "editor.theme": "light" };
+  const settings = { 'editor.theme': 'light' };
+
+  useEffect(() => {
+    const unsubscribe = store.subscribe(() => {
+      const query: string = getQuery(store.getState());
+      const result = query.toLowerCase().includes('collection');
+      if (hasCollection !== result) {
+        setHasCollection(result);
+      }
```

Comment:
doing this 
```ts
setHasCollection(query.toLowerCase().includes('collection'));
```

will cause an infinite re-render.

Reply:
Even if you remove `hasCollection` from the dependencies?

Reply:
actually I can just do this 

```ts
  useEffect(() => {
    const unsubscribe = store.subscribe(() => {
      const query: string = getQuery(store.getState());
      setHasCollection(query.toLowerCase().includes('collection'));
    });
    return unsubscribe;
  }, []);
```

maybe I did something else that caused the infinite re-render.

Reply:
Yes, that should work